app/api/custom_forms.py

```diff
@@ -0,0 +1,105 @@
+from flask_rest_jsonapi import ResourceDetail, ResourceList, ResourceRelationship
+from marshmallow_jsonapi.flask import Schema, Relationship
+from marshmallow_jsonapi import fields
+
+from app.api.bootstrap import api
+from app.api.helpers.utilities import dasherize
+from app.api.helpers.permissions import jwt_required
+from app.models import db
+from app.models.custom_form import CustomForms
+from app.models.event import Event
+from app.api.helpers.db import safe_query
+import marshmallow.validate as validate
+
+
+class CustomFormSchema(Schema):
+    class Meta:
+        type_ = 'custom_form'
+        self_view = 'v1.custom_form_detail'
+        self_view_kwargs = {'id': '<id>'}
+        inflect = dasherize
+
+    id = fields.Integer(dump_only=True)
+    field_identifier = fields.Str(required=True)
+    form = fields.Str(required=True)
+    type = fields.Str(default="text", validate=validate.OneOf(
+        choices=["text", "checkbox", "select", "file", "image"]))
+    is_required = fields.Boolean(default=False)
+    is_included = fields.Boolean(default=False)
+    is_fixed = fields.Boolean(default=False)
+    event = Relationship(attribute='event',
+                         self_view='v1.custom_form_event',
+                         self_view_kwargs={'id': '<id>'},
+                         related_view='v1.event_detail',
+                         related_view_kwargs={'custom_form_id': '<id>'},
+                         schema='EventSchema',
+                         type_='event')
+
+
+class CustomFormList(ResourceList):
+    """
+    Create and List Custom Forms
+    """
+    def query(self, view_kwargs):
+        query_ = self.session.query(CustomForms)
+        if view_kwargs.get('event_id'):
+            event = safe_query(self, Event, 'id', view_kwargs['event_id'], 'event_id')
+            query_ = query_.join(Event).filter(Event.id == event.id)
+        elif view_kwargs.get('event_identifier'):
+            event = safe_query(self, Event, 'identifier', view_kwargs['event_identifier'], 'event_identifier')
+            query_ = query_.join(Event).filter(Event.id == event.id)
+        return query_
+
+    def before_create_object(self, data, view_kwargs):
+        event = None
+        if view_kwargs.get('event_id'):
+            event = safe_query(self, Event, 'id', view_kwargs['event_id'], 'event_id')
+        elif view_kwargs.get('event_identifier'):
+            event = safe_query(self, Event, 'identifier', view_kwargs['event_identifier'], 'event_identifier')
+        if event:
+            data['event_id'] = event.id
+
+    view_kwargs = True
+    decorators = (api.has_permission('is_coorganizer', fetch='event_id',
+                  fetch_as="event_id", model=CustomForms, methods="POST",
+                  check=lambda a: a.get('event_id') or a.get('event_identifier')), )
```

Comment:
Although not an issue with **check**, this is not required. Since there are already two endpoints in route definition so there is no way to get into this resource from any other point not having event_id or event_identifier